packages/snjs/lib/services/api/api_service.ts

```diff
@@ -575,6 +579,82 @@ export class SNApiService extends PureService {
     return response;
   }
 
+  private async tokenRefreshableRequest<T extends MinimalHttpResponse>(
+    params: HttpRequest & { fallbackErrorMessage: string }
+  ): Promise<T> {
+    const preprocessingError = this.preprocessingError();
+    if (preprocessingError) {
+      return preprocessingError as T;
+    }
+    const response: T | HttpResponse = await this.httpService
+      .runHttp(params)
+      .catch((errorResponse: HttpResponse) => {
+        this.preprocessAuthenticatedErrorResponse(errorResponse);
+        if (isErrorResponseExpiredToken(errorResponse)) {
+          return this.refreshSessionThenRetryRequest(params);
+        }
+        return this.errorResponseWithFallbackMessage(
+          errorResponse,
+          params.fallbackErrorMessage
+        );
+      });
+    this.processResponse(response);
+    return response as T;
+  }
+
+  async listSettings(userUuid: UuidString): Promise<ListSettingsResponse> {
+    const resp = await this.tokenRefreshableRequest<ListSettingsResponse>({
```

Comment:
```suggestion
    const response = await this.tokenRefreshableRequest<ListSettingsResponse>({
```

Better for search/readability

Reply:
I did a direct return without an intermediate "resp" variable which I needed for debugging. Thanks for the heads up!